base/sparse/spqr.jl

```diff
@@ -25,122 +25,213 @@ const ORDERING_BESTAMD = Int32(9) # try COLAMD and AMD; pick best#
 const DEFAULT_TOL = Int32(-2) # if tol <= -2, the default tol is used
 const NO_TOL      = Int32(-1) # if -2 < tol < 0, then no tol is used
 
-# for qmult, method can be 0,1,2,3:
-const QTX = Int32(0)
-const QX  = Int32(1)
-const XQT = Int32(2)
-const XQ  = Int32(3)
-
-# system can be 0,1,2,3:  Given Q*R=A*E from SuiteSparseQR_factorize:
-const RX_EQUALS_B    = Int32(0) # solve R*X=B      or X = R\B
-const RETX_EQUALS_B  = Int32(1) # solve R*E'*X=B   or X = E*(R\B)
-const RTX_EQUALS_B   = Int32(2) # solve R'*X=B     or X = R'\B
-const RTX_EQUALS_ETB = Int32(3) # solve R'*X=E'*B  or X = R'\(E'*B)
-
-
 using ..SparseArrays: SparseMatrixCSC
-using ..SparseArrays.CHOLMOD: C_Dense, C_Sparse, Dense, ITypes, Sparse, SuiteSparseStruct, VTypes, common
+using ..SparseArrays.CHOLMOD
 
 import Base: size
-import Base.LinAlg: qrfact
-import ..SparseArrays.CHOLMOD: convert, free!
+import Base.LinAlg: qr, qrfact
 
-struct C_Factorization{Tv<:VTypes} <: SuiteSparseStruct
-    xtype::Cint
-    factors::Ptr{Tv}
-end
+function _qr!(ordering, tol, econ, getCTX, A::Ptr{CHOLMOD.C_Sparse{Tv}},
+    Bsparse::Ptr{CHOLMOD.C_Sparse{Tv}}        = Ptr{CHOLMOD.C_Sparse{Tv}}(C_NULL),
```

Comment:
The indentation here makes it look like these lines are part of the body of the function. An extra indent level for line continuations of block openers would be clearer here IMO.